athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaRecordHandler.java

```diff
@@ -0,0 +1,320 @@
+                    dst.isSet = 1;
+                };
+            case FLOAT8:
+                return (Float8Extractor) (Object context, NullableFloat8Holder dst) ->
+                {
+                    Object value = ((RowContext) context).getNameValue().get(fieldName);
+                    dst.value = Double.parseDouble(value.toString());
+                    dst.isSet = 1;
+                };
+            case DECIMAL:
+                return (DecimalExtractor) (Object context, NullableDecimalHolder dst) ->
+                {
+                    Object value = ((RowContext) context).getNameValue().get(fieldName);
+                    dst.value = new BigDecimal(value.toString());
+                    dst.isSet = 1;
+                };
+            case DATEDAY:
+                return (DateDayExtractor) (Object context, NullableDateDayHolder dst) ->
+                {
+                    Object value = ((RowContext) context).getNameValue().get(fieldName);
+                    dst.isSet = 1;
+                    dst.value =   (int) LocalDate.parse(value.toString()).toEpochDay();
+
+                };
+
+            case DATEMILLI:
+                return (DateMilliExtractor) (Object context, NullableDateMilliHolder dst) ->
+                {
+                    Object value = ((RowContext) context).getNameValue().get(fieldName).toString();
+                    dst.value = LocalDateTime.parse(value.toString()).atZone(BlockUtils.UTC_ZONE_ID).toInstant().toEpochMilli();
+                    dst.isSet = 1;
+                };
+            case VARCHAR:
+                return (VarCharExtractor) (Object context, NullableVarCharHolder dst) ->
+                {
+                    Object value = ((RowContext) context).getNameValue().get(fieldName);
+                    dst.value = value.toString();
+                    dst.isSet = 1;
+                };
+            case VARBINARY:
+                return (VarBinaryExtractor) (Object context, NullableVarBinaryHolder dst) ->
+                {
+                    Object value = ((RowContext) context).getNameValue().get(fieldName).toString();
+                    dst.value =  value.toString().getBytes();
+                    dst.isSet = 1;
+                };
+            default:
+                throw new RuntimeException("Unhandled type " + fieldType);
+        }
+    }
+
+    private static class RowContext
+    {
+
+        private final String queryId;
+        private HashMap<String, Object> nameValue;
+
+        public RowContext(String queryId){
+            this.queryId = queryId;
+        }
+
+        public void setNameValue(HashMap<String, Object> map){
+            this.nameValue = map;
+        }
+        public HashMap<String, Object> getNameValue() {
+            return this.nameValue;
+        }
+    }
+
+
+    /*
+    Method to create the Parquet read request
+     */
+    private static SelectObjectContentRequest generateBaseParquetRequest(String bucket, String key)
```

Comment:
Is this using S3 select? I dont think we should use S3 select, especially since you are doing a Select *... are we doing this just to avoid having parquet dependencies in our code? What format does S3 select use for results?

Reply:
Yes. Using S3 Select to avoid the parquet/avro/hadoop dependencies. Including these causes the app size > 50MB and also causes dependency conflicts. Still researching a way to read parquet files natively.

Reply:
Did we make any progress on this? If S3 select is our only option then I suppose we can simply note it in the docs. Is the customer you are working with using the connector with S3 Select too?

Reply:
Currently, S3 select is the only option. I have mentioned this in the Readme file. Yes, customer is also using S3 Select.

Reply:
We have a new option that was just released in lake formation which is more appropriate than S3 Select. It even returns Apache Arrow format already. We can look at moving to that post GA. Its currently in Preview only.  I've raised an issue, #325 , so we dont forget this.